src/isVisible.ts

```diff
@@ -0,0 +1,5 @@
+export default function isVisible(
+  node: HTMLElement | null
+) : boolean {
+  return node ? !!(node.offsetWidth || node.offsetHeight || node.getClientRects().length) : false;
```

Comment:
do you not need to check if `display` is something other than `none`? or opacity and visibility for that matter

Reply:
This code was extracted from the code of the :visible pseudo-selector implementation in jQuery, even in the latest release 3.6.0. I'd say that testing for style display is not part of it because it would require to use getComputedStyle which can possibly be slower than reading the element's dimensions. I can look for benchmarks on this topic as well.

Reply:
I think this is fine, It's more that "visible" here is a bit of a misnomer but still a useful helper as is